src/components/Segment.jsx

```diff
@@ -91,24 +121,71 @@ function UpdateSegmentPopup(segment, segmentName, distance, onUpdateSegment, onD
   );
 }
 
-const Segment = ({ segment, segmentName, distance, onUpdateSegment, onDeleteSegment }) => {
+// Custom component to render Name
+const Name = ({ segment, segmentName, segmentLocation }) => {
+  return (
+    <>
+      <VStack>
+        <div className="segmentname-container">
+          {segment} {segmentName}
+        </div>
+        <div className="location-container">{segmentLocation}</div>
+      </VStack>
+    </>
+  );
+};
+
+const Segment = ({
+  segment,
+  segmentName,
+  segmentLocation,
+  link,
+  parking,
+  distance,
```

Comment:
⚠️ **[eslint]** <[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'distance' is defined but never used.